libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py

```diff
@@ -372,19 +379,36 @@ def _generate(
             # set param to `functions` until core tool/function calling implemented
             raw_tools = params.pop("functions") if "functions" in params else None
             tools = _format_tools_to_vertex_tool(raw_tools) if raw_tools else None
-            response = chat.send_message(
-                message,
-                generation_config=params,
-                tools=tools,
-                safety_settings=safety_settings,
-            )
-            generations = [
-                ChatGeneration(
-                    message=_parse_response_candidate(c),
-                    generation_info=get_generation_info(c, self._is_gemini_model),
+            try:
+                response = chat.send_message(
+                    message,
+                    generation_config=params,
+                    tools=tools,
+                    safety_settings=safety_settings,
                 )
-                for c in response.candidates
-            ]
+            except ResponseBlockedError as e:
+                response = e.responses[0]
+            generations = []
+            for c in response.candidates:
+                if c.finish_reason in [
+                    gapic_content_types.Candidate.FinishReason.STOP,
+                    gapic_content_types.Candidate.FinishReason.FINISH_REASON_UNSPECIFIED,
+                ]:
+                    generations.append(
+                        ChatGeneration(
+                            message=_parse_response_candidate(c),
+                            generation_info=get_generation_info(
+                                c, self._is_gemini_model
+                            ),
+                        )
+                    )
+                else:
+                    error_str = "Text was found that violates the safety policy"
```

Comment:
maybe let's throw an exception with `finish_reason`?

Reply:
The whole point of this pull request is to be able to not throw an exception to have a similarly behaviour to `OpenAIModerationChain`. VertexAI's opt-out approach to safety features, in contrast to OpenAI's opt-in approach.